projects/com.oracle.truffle.llvm.nodes.impl/src/com/oracle/truffle/llvm/nodes/impl/op/compare/LLVMFloatCompareNode.java

```diff
@@ -54,42 +54,71 @@ private static boolean areOrdered(float... vals) {
     public abstract static class LLVMFloatOltNode extends LLVMFloatCompareNode {
         @Specialization
         public boolean executeI1(float val1, float val2) {
-            return areOrdered(val1, val2) && val1 < val2;
+            if (val1 < val2) {
+                assert areOrdered(val1, val2);
+                return true;
+            } else {
+                return false;
+            }
```

Comment:
If the comparison is used in an branch, then the branch probability profiling that is already injected in the successor selection should suffice. However, the value could also be stored into the frame or somewhere else, and then used at a different location. Do you think that branch probability injection could improve performance in this case?

/cc @lukasstadler @gilles-duboscq